src/com/mendix/widget/StarRating/components/StarRating.ts

```diff
@@ -0,0 +1,85 @@
+import { Component, createElement, DOM } from "react";
+
+import * as Rating from "react-rating";
+import { Alert } from "./Alert";
+
+export interface StarRatingProps {
+    initialRate?: number;
+    handleOnChange?: Function;
+    onRateMicroflow?: string;
+    ownerGUID?: string;
+    readOnly: boolean;
+    rateType: "rating" | "average";
+    configurationError?: string;
+}
+
+export interface StarRatingState {
+    errorMessage: string;
+}
+
+export class StarRating extends Component<StarRatingProps, StarRatingState> {
+    private fractions: number;
+    private rateType: string;
+    private start: number;
+    private step: number;
+    private stop: number;
+
+    constructor(props: StarRatingProps) {
+        super(props);
+
+        this.state = { errorMessage: "" };
+    }
+
+    render() {
+        this.fractions = 2;
+        this.start = 0;
+        this.step = 1;
+        this.stop = 5;
+        this.rateType = this.props.rateType;
```

Comment:
Yes, that was the intention for its use.
no need to `const rateType` in the other functions.